grails-app/utils/org/pih/warehouse/DateFormatName.groovy

```diff
@@ -0,0 +1,21 @@
+package org.pih.warehouse
+
+// Enum for storing labels used in dates formatting.
+// In case of adding new format we have to:
+// 1. Add new format in message.properties file
+// 2. Add the newly created label to this enum
+// This enum is used in:
+// 1. G tag - <g:formatDate date={your date} formatName={id of enum property} />
+// 2. LocalizationUtil.formatDate(your date, id of enum property)
+
+enum DateFormatName {
+    DEFAULT('default.date.format'),
+    CUSTOM('custom.date.format'),
+    EXPIRY('expiry.date.format'),
```

Comment:
I'm not sure if this is the types we want. DEFAULT and CUSTOM are more generic sounding while EXPIRY is for a very specific purpose (product expiration). It's mixing and matching a bit.

What are you imagining is the behaviour if we add a new feature X that also needs to display only month and year? Would it re-use the EXPIRY type or would it need to create its own X type that also formats MMM/yyyy?

My instinct tells me that we'd be better off with more directly descriptive types (ex: DATE_TIME, FULL_DATE. MONTH_YEAR) but let me know your thoughts

Reply:
and if you want to avoid having DEFAULT be in the enum itself you could move it to a constant in the enum:

```
public final DateFormatName DEFAULT = DATE_TIME
```